### PR TITLE
Code refactoring

### PR DESCRIPTION
**Pull request by LunyaticCat**
## Short roundup of the initial problem
All files are currently in the root directory, making it really hard for new devs to join the project.

## What will change with this Pull Request?
every cockatrice files are put into different folders matching what they do.

### CODE REVIEW

File: cockatrice/src/card_database_parser/cockatrice_xml3.cpp
Comment (BruebachL) [inline cockatrice/src/card_database_parser/cockatrice_xml3.cpp:1]: Maybe add another underscore for readability (e.g. cockatrice_xml_3.cpp)

File: cockatrice/src/card_database_parser/cockatrice_xml4.cpp
Comment (BruebachL) [inline cockatrice/src/card_database_parser/cockatrice_xml4.cpp:1]: Maybe add another underscore for readability (e.g. cockatrice_xml_3.cpp)

File: cockatrice/src/dialogs/dlg_connect.cpp
Comment (BruebachL) [inline cockatrice/src/dialogs/dlg_connect.cpp:1]: Some unwanted changes to UI naming here.

**Review by BruebachL — CHANGES_REQUESTED** [review]
LGTM, pending changes.

**Comment by tooomm** [comment]
Where does the 10k reduced lines come from?

**Comment by LunyaticCat** [comment]
> Where does the 10k reduced lines come from?

probably from github thinking I deleted a file called "cardzone" and added a file called "card_zone" when I simply renamed and moved it like other files

**Comment by ZeldaZach** [comment]
Can you remove the oracle/ translations as well?

**Comment by ZeldaZach** [comment]
Can you resolve the merge conflicts?

File: cockatrice/src/client/tabs/tab_supervisor.cpp
Comment (tooomm) [inline cockatrice/src/client/tabs/tab_supervisor.cpp:17]: Not sure about this one

**Review by tooomm — CHANGES_REQUESTED** [review]
Great, the list of changes is now somewhat manageable. Thanks.
I had a look and put some notes when skimming trough.

Others, if you want to look at the source repo and have a look at the new structure, check if something should be changed: https://github.com/LunyaticCat/Cockatrice/tree/code-refractoring

File: cockatrice/src/dialogs/dlg_register.cpp
Comment (LunyaticCat) [reply cockatrice/src/dialogs/dlg_register.cpp:354]: Ah yes, refactoring must have changed it without me noticing ^^'

**Comment by tooomm** [comment]
@LunyaticCat It's now only about the conflicts that need resolution 👍 

image

**Review by tooomm — APPROVED** [review]
Thanks for resolving the conflicts! 👍 

Will defer to others to sanity check the actual naming and folder structure. @ZeldaZach @ebbit1q

**Review by ZeldaZach — APPROVED** [review]
If you can untrack the changes on the `.github` and `webclient` dirs, this is good to go

**Comment by ZeldaZach** [comment]
Can you resolve conflicts one more time? Then I'll merge this through!

**Comment by LunyaticCat** [comment]
> Can you resolve conflicts one more time? Then I'll merge this through!

ok I think I did it!

**Pull request merged**
